repos/system_upgrade/common/actors/targetuserspacecreator/libraries/userspacegen.py

```diff
@@ -754,6 +813,71 @@ def _create_target_userspace(context, packages, files, target_repoids):
         rhsm.set_container_mode(target_context)
 
 
+def install_target_rhui_client_if_needed(context, indata):
+    if not indata.rhui_info:
+        return
+
+    target_major_version = get_target_major_version()
+    userspace_dir = _get_target_userspace()
+    _create_target_userspace_directories(userspace_dir)
+
+    setup_info = indata.rhui_info.target_client_setup_info
+    if setup_info.preinstall_tasks:
+        preinstall_tasks = setup_info.preinstall_tasks
+
+        for file_to_remove in preinstall_tasks.files_to_remove:
+            context.remove(file_to_remove)
+
+        for copy_info in preinstall_tasks.files_to_copy_in:
+            context.makedirs(os.path.dirname(copy_info.dst), exists_ok=True)
+            context.copy_to(copy_info.src, copy_info.dst)
+
+    # @Todo: We might need to handle GPG stuff here in order to install the client
```

Comment:
Under what circumstances we would be handling the GPG stuff?

Reply:
If the repo definition contained `gpgkey=...` that is available only on target rhel.